phac-nml/irida-next
# Data Export: Add Workflow Executions UI

## What does this PR do and why?
This PR adds the UI to create analysis (workflow execution) exports through the IRIDA Next website.

## Screenshots or screen recordings
[Screencast from 2024-04-16 09:23:58 AM.webm](https://github.com/phac-nml/irida-next/assets/82407232/1e0ddd57-c080-4839-9a8f-7e74412b1346)

## How to set up and validate locally
1. Create a workflow execution as described in [PR507](https://github.com/phac-nml/irida-next/pull/507)
2. While its state is not completed, check that the create export button is disabled on the workflow execution page
3. Once completed, create an export
4. After 30s, ensure the preview file structure is as expected, and download the export to ensure the zip contents are also as expected.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
